cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -3698,15 +3740,16 @@
           (result Xmm (vec_insert_lane $F32X4 result a3 3))
         )
         result))
-(rule (x64_round $F64X2 a imm)
+(rule (x64_round $F64X2 (RegMem.Reg a) imm)
```

Comment:
Not sure I completely understand: this seems to be saying that we expect `a` to already be in a register. So I assume all the lowering rules already do this and this is just an extra check that this is the case?

Reply:
Ah so before it was `a: Value` but I called this helper from another location which only had a register so I changed it to `a: RegMem`. This meant that anything coerced into memory had to be explicitly lifted from memory (the last rule here) so the base rules like making calls only worked with registers.